tests/test_dates.py

```diff
@@ -0,0 +1,14 @@
+import unittest
+import time
+from elifecrossref import dates
+
+
+class TestIsoDateString(unittest.TestCase):
+
+    def test_iso_date_string(self):
+        date_string = '2020-06-03'
+        pub_date = time.strptime(date_string, '%Y-%m-%d')
+        self.assertEqual(dates.iso_date_string(pub_date), date_string)
```

Comment:
Not a good test. What are we testing here? That Python is behaving as expected? If so, then where does one stop? Python is already well tested (we hope). What we're actually testing here is the `dates.iso_date_string` function. We give it input and assert the output is as expected.

```python
pub_date = (2019, 12, 31, 0, 0, 0, 1, 365, -1)
expected = '2019-12-31'
self.assertEqual(expected, dates.iso_date_string(pub_date))
```

How we got to the point where `pub_date` is a tuple or time object, or that you can get there using `time.strptime` is irrelevant. `dates.iso_date_string` is expected to receive `pub_date` as input and the expected output is a YMD string.

If you have some other custom function that creates values suitable for input to `dates.iso_date_string` you could use that.

Reply:
I just added this date test to see how closely you're reading the code :)

Beyond that, I think what happened is I needed an ISO data string for inclusion in the XML, and so I quickly put together a separate function, but I based it on using `str.format()`. Later, I remembered the dates in the `Article()` object that is returned when getting the pub date from it is a `time.struct_time` object, and then I switched over to using a more stanard date formatting `time.strftime()`.

The test case that remains is not very useful, as you say, outside of checking for any typos in the `dates.iso_date_string()` function. The code is covered by other tests, though, so it's not required to test it even for code coverage.

Wrapping up populating an `Article()` with dates and then testing this date function doesn't add much either I think, because those are tested already in the `elife-article` library.

In summmary, I think this test case can be deleted! Thanks for checking so carefully.